tasks/task_manager.hh

```diff
@@ -346,9 +356,29 @@ public:
             id = task->id();
 
             if (parent_d) {
-                co_await _tm.container().invoke_on(parent.get_owner_shard(), [task = std::move(parent), child = std::move(task)] (task_manager& tm) mutable {
-                    task->add_child(std::move(child));
-                });
+                co_await add_child(std::move(parent), std::move(task));
+            }
+            co_return id;
+        }
+
+        // Responsibility for task_impl_ptr correctness is put on caller.
+        future<task_id> make_task(unsigned shard, task::task_impl_ptr task_impl_ptr, parent_data parent_d = parent_data{}) {
+            foreign_task_ptr parent;
+            uint64_t sequence_number = 0;
+            if (parent_d) {
+                parent = co_await get_parent(parent_d);
+                sequence_number = parent->get_sequence_number();
+            }
+
+            auto task = co_await _tm.container().invoke_on(shard, [task_impl_ptr = std::move(task_impl_ptr), sequence_number] (task_manager& tm) mutable {
+                auto task = make_lw_shared<task_manager::task>(std::move(task_impl_ptr));
+                task->get_status().sequence_number = sequence_number == 0 ?  task->get_module()->new_sequence_number() : sequence_number;
```

Comment:
I have no clue what I tried to do here, but this function does not make any sense on many layers...

I am rewriting it from the very beginning. The function would be expected to be called from the child's shard and will return task_ptr. The old `make_task` will still return `task_id`, though, since it can be called from any shard.